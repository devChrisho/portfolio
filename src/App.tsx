import { useEffect, useMemo } from "react";
import "./App.css";
import {
  Box,
  CssBaseline,
  ThemeOptions,
  ThemeProvider,
  createTheme,
} from "@mui/material";
import { useSelector } from "react-redux";
import { selectSystem } from "./redux/selectors/systemSelectors";
import themeOptionsGenerator from "./theme/themeOptions";
import TopAppBar from "./components/TopAppBar";

function App() {
  // Redux states
  const { isDarkMode } = useSelector(selectSystem);

  const themeOptions: ThemeOptions = useMemo(() => {
    return themeOptionsGenerator({ isDarkMode });
  }, [isDarkMode]);

  const theme = useMemo(() => createTheme(themeOptions), [themeOptions]);

  useEffect(() => {
    (window as any).theme = theme;
  }, [theme]);

  return (
    <>
      <ThemeProvider theme={theme}>
        <CssBaseline>
          <Box sx={{ flexGrow: 1, height: "100vh" }}>
            <TopAppBar />
          </Box>
        </CssBaseline>
      </ThemeProvider>
    </>
  );
}

export default App;
