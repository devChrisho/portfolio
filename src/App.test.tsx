import { render } from "@testing-library/react";
import App from "./App";
import { Provider } from "react-redux";
import store from "./redux/store";

describe("App unit test", () => {
  it("should pass snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
