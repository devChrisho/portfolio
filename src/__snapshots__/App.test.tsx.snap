// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App unit test > should pass 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-znyx2w-MuiPaper-root"
  >
    testing
  </div>
  <div>
    <a
      href="https://vitejs.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="/vite.svg"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="/src/assets/react.svg"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1fwhrvi-MuiTypography-root"
    >
      Hello world a
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-dwnzdl-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Click me
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button>
      count is 
      0
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</div>
`;

exports[`App unit test > should pass snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-znyx2w-MuiPaper-root"
  >
    testing
  </div>
  <div>
    <a
      href="https://vitejs.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="/vite.svg"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="/src/assets/react.svg"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1fwhrvi-MuiTypography-root"
    >
      Hello world a
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-dwnzdl-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Click me
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button>
      count is 
      0
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</div>
`;
